python/ql/lib/semmle/python/dataflow/new/internal/DataFlowPublic.qll

```diff
@@ -540,6 +540,35 @@ class GuardNode extends ControlFlowNode {
 }
 
 /**
+ * Holds if the guard `g` validates `node` upon evaluating to `branch`.
+ *
+ * The expression `e` is expected to be a syntactic part of the guard `g`.
+ * For example, the guard `g` might be a call `isSafe(x)` and the expression `e`
+ * the argument `x`.
+ */
+signature predicate guardChecksSig(GuardNode g, ControlFlowNode node, boolean branch);
+
+/**
+ * Provides a set of barrier nodes for a guard that validates a node.
+ *
+ * This is expected to be used in `isBarrier`/`isSanitizer` definitions
+ * in data flow and taint tracking.
+ */
+module BarrierGuard<guardChecksSig/3 guardChecks> {
+  /** Gets a node that is safely guarded by the given guard check. */
+  ExprNode getABarrierNode() {
```

Comment:
@yoff should this be ExprNode or CfgNode? It seems the only difference is that CfgNodes can also be patterns from a match statement. I simply don't know whether such patterns should be eligible for being the result of a guards check :thinking: Anyway, I was just a bit suspicious about this since we use ExprNode very little in our code, and thought it might just be a leftover from `@aschackmull` normally using that for Java.

https://github.com/github/codeql/blob/02614e1c7fd7d214fd6c100829e0a36575ab9630/python/ql/lib/semmle/python/dataflow/new/internal/DataFlowPublic.qll#L27-L32

https://github.com/github/codeql/blob/02614e1c7fd7d214fd6c100829e0a36575ab9630/python/ql/lib/semmle/python/dataflow/new/internal/DataFlowPublic.qll#L267-L268

Reply:
Feel free to change it as a follow-up. I just kept the status quo in this case.